shell.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import(
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"regexp"
+)
+
+// ShellCommand is a helper function that executes the specified shell command with its
+// specified arguments. It prints the output of the shell command.
+func ShellCommand(cmd string, args ...string) (out string, err error) {
+	var output []byte
+
+	output, err = exec.Command(cmd, args...).CombinedOutput()
+	if err != nil {
+		return "", err
+	}
+
+	return string(output), err
+}
+
+// Clone will clone the specified branch from the repository in path 
+// and switch to its directory.
+func Clone(path string, branch string) (err error) {
```

Comment:
Perhaps better would be to accept an org and a repo, or perhaps to accept an entire valid url for git checkout, and parse that to calculate a checkout directory.